lib/rules/no-native-reassign.js

```diff
@@ -17,6 +18,8 @@ module.exports = {
             recommended: true
         },
 
+        deprecated: true,
```

Comment:
I put it there because it's what other suggested and what @mysticatea implemented in https://github.com/eslint/eslint/pull/6756. That's an example of something which is not strictly documentation. That said, I don't have a problem to move it. If we add `replaceBy`, it might make more sense to keep them together in `docs`.

I should probably keep `replacedBy` as an array, just in case a rule is replaced by many.